### PR TITLE
TST: remove nose

### PR DESCRIPTION
**Pull request by jreback**
TST: raise nose.SkipTest -> pytest.skip
TST: remove KnownFailure (unused), should be replaced by pytest.xfail anyhow

xref #15341

### CODE REVIEW

File: pandas/tests/test_expressions.py [hunk 1]
```diff
@@ -21,13 +21,7 @@
...
 
 if not expr._USE_NUMEXPR:
-    try:
-        import numexpr  # noqa
-    except ImportError:
-        msg = "don't have"
-    else:
-        msg = "not using"
-    raise nose.SkipTest("{0} numexpr".format(msg))
+    numexpr = pytest.importorskip('numexpr')
```
Comment (jorisvandenbossche) [inline pandas/tests/test_expressions.py]: Didn't look into if this is important, but the importskip here is not fully equivalent with the above.

File: pandas/util/testing.py [hunk 2]
```diff
@@ -2015,10 +2015,6 @@ def package_check(pkg_name, version=None, app='pandas', checker=LooseVersion,
     checker : object, optional
         The class that will perform the version checking.  Default is
         distutils.version.LooseVersion.
-    exc_failed_import : Exception, optional
-        Class of the exception to be thrown if import failed.
-    exc_failed_check : Exception, optional
-        Class of the exception to be thrown if version check failed.
```
Comment (jorisvandenbossche) [inline pandas/util/testing.py]: also remove the actual arguments?

File: pandas/util/testing.py [hunk 3]
```diff
@@ -2042,40 +2039,31 @@ def package_check(pkg_name, version=None, app='pandas', checker=LooseVersion,
...
     *kwargs Keyword parameters passed to `package_check`
     """
-    from nose import SkipTest
-    package_check(exc_failed_import=SkipTest,
-                  exc_failed_check=SkipTest,
-                  *args, **kwargs)
+    package_check(*args, **kwargs)
 
 
 def skip_if_no_package_deco(pkg_name, version=None, app='pandas'):
```
Comment (jorisvandenbossche) [inline pandas/util/testing.py]: remove this whole function? It's nowhere used, and you removed its tests

**Review by jorisvandenbossche — APPROVED** [review]
Looks good!

Added some minor comments.

File: pandas/util/testing.py (hunk 3 above)
Comment (jreback) [inline pandas/util/testing.py]: yeah I don't know how to actually test if a tests is actually skipped; in nose you can catch the ``SkipTest``, but for pytest you would never do this.

Comment (jreback) [inline pandas/util/testing.py]: yeah, removing it. We have lots of testing cruft that pytest just does for free.

File: pandas/util/testing.py (hunk 2 above)
Comment (jreback) [inline pandas/util/testing.py]: done

File: pandas/tests/test_expressions.py (hunk 1 above)
Comment (jreback) [inline pandas/tests/test_expressions.py]: yeah this is the only way to actually skip the entire module (well you *can* simply use a skipif on the classes, but this is cleaner).

Comment (jorisvandenbossche) [inline pandas/tests/test_expressions.py]: I meant that previously the test was also skipped if _USE_NUMPEXPR is false, now only when the import fails

Comment (jreback) [inline pandas/tests/test_expressions.py]: _USE_NUMEXPR == can import numexpr (and its not turned off)

**Pull request closed without merging**